src/builder.rs

```diff
@@ -70,26 +70,13 @@ pub enum BuildError {
     CircuitError(#[from] circuit_builder::CircuitBuildError),
 }
 
-impl AsMut<HugrMut> for HugrMut {
-    fn as_mut(&mut self) -> &mut HugrMut {
-        self
-    }
-}
-impl AsRef<HugrMut> for HugrMut {
-    fn as_ref(&self) -> &HugrMut {
-        self
-    }
-}
-
-/// Trait allowing treating type as (im)mutable reference to [`HugrMut`]
-pub trait HugrMutRef: AsMut<HugrMut> + AsRef<HugrMut> {}
-impl HugrMutRef for HugrMut {}
-impl HugrMutRef for &mut HugrMut {}
+/// Trait allowing treating type as (im)mutable reference to [`Hugr`]
+pub trait HugrMutRef: AsMut<Hugr> + AsRef<Hugr> {}
+impl<H: HugrMut> HugrMutRef for H {}
+impl HugrMutRef for &mut Hugr {}
```

Comment:
Can we implement `AsMut` and `AsRef` instead?

Reply:
At present HugrMutRef is a trait that one implements *on top of* AsMut and AsRef. AFAICS, the only purpose is to make it easy to parameterize (i.e. over `T: HugrMutRef` rather than `T: AsMut<Hugr> + AsRef<Hugr>`).

So, removed via bulk replace. (I have not gone through and checked whether any of them can be just `: AsMut<Hugr>` or `: AsRef<Hugr>` rather than `+` of both.)

This is a bit messy textually even if I think makes sense. (Sadly you can't define `type HugrMutRef = AsRef<Hugr> + AsMut<Hugr>` - `type` must really be a *type* not merely a *bound*.)

However, have kept in separate commit b48e2f6a7175edb68bf7e4b14b34aa72a0275adb in case you think this is too messy...

Reply:
@aborgna-q can you have a quick look at that b48e2f6a7175edb68bf7e4b14b34aa72a0275adb in case you think better of it....